src/sentry/integrations/gitlab/repository.py

```diff
@@ -14,20 +14,18 @@ class GitlabRepositoryProvider(providers.IntegrationRepositoryProvider):
 
     def get_installation(self, integration_id, organization_id):
         if integration_id is None:
-            raise ValueError('%s requires an integration_id' % self.name)
+            raise IntegrationError('%s requires an integration id.' % self.name)
 
-        try:
-            integration_model = Integration.objects.get(
-                id=integration_id,
-                organizations=organization_id,
-            )
-        except Integration.DoesNotExist as error:
-            self.handle_api_error(error)
+        integration_model = Integration.objects.get(
+            id=integration_id,
+            organizations=organization_id,
+            provider='gitlab',
```

Comment:
If you have the primary key value here, will the provider condition add much?

Reply:
No, but I don't think it hurts anything to be cautious. This value comes from the api, and there is no check beforehand that the provider for the given integration id is the provider that the api was given. Same with organization, there's no check that org is associated with that integration until this point.

Reply:
Fair enough.